### ISSUE TITLE
Support http authentication in parasol

### ISSUE THREAD

**Issue by acabrol** [alx/parasol#13]
Add "withCredentials" parameters on http ajax queries to support http basic auth through Reverse Proxy.

` $http({
        url: 'url of service',
        method: "POST",
        data: {test :  name },
        withCredentials: true,
        headers: {
                    'Content-Type': 'application/json; charset=utf-8'
        }
    });`

https://stackoverflow.com/questions/21455045/angularjs-http-cors-and-http-authentication

**Comment by alx** [comment]
There's a new *fetchOptions* parameter to  to allow sending credentials to other domains with CORS : https://github.com/github/fetch#sending-cookies

Here is a setting example with LoaderJson:

```
    <div
      class="parasol"
      data-settings='{
        "networks": [
          {
            "url": "/docs/assets/data/les-miserables.json",
            "fetchOptions": {"credentials": "include"},
            "name": "LDA for movie reviews",
            "options": {
              "layout": "forcelink",
```

Let me know if working and if issue can be closed.

**Comment by acabrol** [comment]
**# html setting:**

`data-settings='{
        "networks": [ { "url": "/api/graph", "name": "dfm", "fetchOptions": {"credentials": "include"}, "options": { "layout": "forcelink", "loader": { "name": "json" } } } ],
        "ui": { "drawers": [ { "id": "left", "open": true, "components": [ {"name": "Legend"}, {"name": "Divider"}, {"name": "SearchInput"}, {"name": "NetworkList"}, {"name": "Divider"}, {"name": "ForceLinkSettings"} ] }, { "id": "right", "open": true, "openSecondary": true, "components": [ {"name": "AppBar"}, {"name": "Divider"}, {"name": "SelectedNode"} ] } ] }
      }'`
**# apache setting:**
`Header set Access-Control-Allow-Credentials "true"
Header set Access-Control-Allow-Origin "*"
Header set Access-Control-Allow-Methods "GET, POST, PUT, DELETE, OPTIONS"
Header set Access-Control-Allow-Headers "Origin, X-Requested-With, Content-Type, Accept, Authorization"`

**# result:**
`<html><head>
<title>401 Unauthorized</title>
</head><body>
<h1>Unauthorized</h1>
<p>This server could not verify that you
are authorized to access the document
requested.  Either you supplied the wrong
credentials (e.g., bad password), or your
browser doesn't understand how to supply
the credentials required.</p>
<hr>
<address>Apache  at dfm </address>
</body></html>`

**Comment by alx** [comment]
fixed in 548cd31
